### PR TITLE
v1 Study session testing and refinement

### PR DESCRIPTION
**Pull request by jltha**
Mainly edited the Ui, fixed some bugs in study session, and added junit tests/asserts.

Also made very small changes in other sections of code:
1) shifting of private constants to Message class (Storage)
2) wrapping lines that are getting too long (Parser prepareAdd)
3) shifting some methods up for greater clarity and changed phrasing of messages to a friendlier tone (Related to Parser prepareAdd)

### CODE REVIEW

**Review by thedarie — APPROVED** [review]

**Review by laiisaac — APPROVED** [review]
Thanks for correcting some code quality issues with Timer and TimerLogic!

File: src/main/java/seedu/sherpass/util/Parser.java [hunk 1]
```diff
@@ -103,12 +99,54 @@ private static void printMissingInputMessage() {
...
+        Ui anotherUi = new Ui();
+        anotherUi.showToUser("It seems that the date and time\nyou gave is not in the correct format.\n"
+                + "Would you like to re-enter a valid date and time? (Y/N)\n"
+                + "*Enter 'No' to skip the adding of this task*");
+        anotherUi.showLine();
+        while (true) {
+            String input = anotherUi.readCommand();
+            anotherUi.showLine();
+            if (input.trim().equalsIgnoreCase("Y")
+                    || input.trim().equalsIgnoreCase("Yes")) {
```
Comment (BradenTeo) [inline src/main/java/seedu/sherpass/util/Parser.java:112]: Possible to put lines 111 and 112 on the same line?

File: src/main/java/seedu/sherpass/util/Parser.java [hunk 2]
```diff
@@ -103,12 +99,54 @@ private static void printMissingInputMessage() {
...
+                anotherUi.showToUser("Understood. Please key in the date and time you wish to save.");
+                anotherUi.showLine();
+                anotherUi.showToUser("Enter valid date input:");
+                input = anotherUi.readCommand();
+                anotherUi.showLine();
+                return prepareTaskDate(input.trim());
+            }
+            if (input.trim().equalsIgnoreCase("N")
+                    || input.trim().equalsIgnoreCase("No")) {
```
Comment (BradenTeo) [inline src/main/java/seedu/sherpass/util/Parser.java:121]: Can be on the same line too?

**Review by BradenTeo — APPROVED** [review]

File: src/main/java/seedu/sherpass/util/Parser.java (hunk 1 above)
Comment (jltha) [reply src/main/java/seedu/sherpass/util/Parser.java:112]: Lines were wrapped as it was nearly reaching the 120 character limit.

File: src/main/java/seedu/sherpass/util/Parser.java (hunk 2 above)
Comment (jltha) [reply src/main/java/seedu/sherpass/util/Parser.java:121]: Same situation as above

**Pull request merged**